crates/rover-client/src/headers.rs

```diff
@@ -0,0 +1,32 @@
+use crate::RoverClientError;
+use std::env;
+
+const CONTENT_TYPE: &str = "application/json";
+const CLIENT_NAME: &str = "rover-client";
+
+/// Function for building a [HeaderMap] for making http requests.
+///
+/// Takes a single argument, "api_key"m and returns a [HeaderMap].
+pub fn build(api_key: &str) -> Result<reqwest::header::HeaderMap, RoverClientError> {
+    let mut headers = reqwest::header::HeaderMap::new();
+
+    let content_type =
+        reqwest::header::HeaderValue::from_str(CONTENT_TYPE).expect("is valid header");
+    headers.insert("Content-Type", content_type);
+
+    let client_name = reqwest::header::HeaderValue::from_str(CLIENT_NAME).expect("is valid header");
+    headers.insert("apollographql-client-name", client_name);
+
+    let client_version = reqwest::header::HeaderValue::from_str(
+        &env::var("CARGO_PKG_VERSION").unwrap_or_else(|_| String::from("unknown")),
+    )
+    .expect("is valid header");
+    headers.insert("apollographql-client-version", client_version);
```

Comment:
what are these headers required for? is it documented/can we add comments explaining?